arangod/Aql/AqlItemBlock.cpp

```diff
@@ -295,34 +295,23 @@ void AqlItemBlock::destroy() noexcept {
   try {
     _shadowRowIndexes.clear();
 
-    if (_valueCount.empty()) {
-      eraseAll();
-      rescale(0, 0);
-      TRI_ASSERT(numEntries() == 0);
-      return;
-    }
-
-    size_t const n = numEntries();
-    for (size_t i = 0; i < n; i++) {
-      auto& it = _data[i];
-      if (it.requiresDestruction()) {
-        auto it2 = _valueCount.find(it);
-        if (it2 != _valueCount.end()) {  // if we know it, we are still responsible
-          TRI_ASSERT((*it2).second > 0);
-
-          if (--((*it2).second) == 0) {
-            decreaseMemoryUsage(it.memoryUsage());
-            it.destroy();
-            _valueCount.erase(it2);
-          }
-        }
+    size_t totalUsed = 0;
+    for (auto& it : _valueCount) {
+      auto const& count = it.second;
+      if (ADB_LIKELY(count > 0)) {
+        auto value = it.first;
+        value.destroy();
+        totalUsed += value.memoryUsage();
```

Comment:
shouldn't these two lines be inversed? how can we tell the `value`'s memoryUsage otherwise?

Reply:
🤦